docs/developer-env.md

```diff
@@ -0,0 +1,473 @@
+(developer-env)=
+
+# Environment setup
+
+Setting up a Nextflow development environment is a prerequisite for creating, testing, and optimizing data analysis pipelines. The steps below outline recommended tools for setting up an optimal Nextflow development environment.
+
+:::{note}
+If you are using a Windows computer, you first need to install and configure the Windows Subsystem for Linux (WSL). See {ref}`wsl` for installation instructions.
+:::
+
+(vscode-install)=
+
+## VS Code
+
+Installing an Integrated Development Environment (IDE) is an essential step for setting up your environment and provide a user-friendly interface for writing, editing, and managing code.
+
+Visual Studio Code (VS Code) is a popular lightweight IDE that is known for its versatility and extensibility. It offers features like syntax highlighting, intelligent code completion, and integrated debugging tools for various programming languages. VS Code supports Windows, macOS, and Linux, and is a good choice for both new and experienced Nextflow developers.
+
+````{tabs}
+
+```{group-tab} Windows
+
+To install VS Code on Windows:
+
+1. Visit the [VS Code](https://code.visualstudio.com/download) website.
+1. Download VS Code installer for Windows.
+1. Double-click the installer executable (`.exe`) file and follow the step-by-step setup installation wizard.
+
+```
+
+```{group-tab} macOS
+
+To install VS Code on macOS:
+
+1. Visit the [VS Code](https://code.visualstudio.com/download) website.
+1. Download VS Code for macOS.
+1. Drag the `Visual Studio Code.app` application to the Applications folder to make it available in the macOS Launchpad.
+
+```
+
+```{group-tab} Linux
+
+To install VS Code on Linux Debian/Ubuntu distributions:
+
+1. Visit the [VS Code](https://code.visualstudio.com/download) website.
+1. Download VS Code Linux Debian/Ubuntu (`.deb`) distribution.
+1. Open a new terminal window.
+1. Navigate to the folder containing your VS Code download.
+1. Run `sudo apt install ./<file>.deb`, replacing `<file>` with the full file name.
+   
+   :::{note}
+   If you're using an older Linux distribution, run `sudo dpkg -i <file>.deb` to install VS Code and `sudo apt-get install -f` to install dependencies.
+   :::
+   
+See [Linux installation](https://code.visualstudio.com/docs/setup/linux#_installation) for information about installing VS Code on other distributions.
+
+```
+
+````
+
+## Extensions
```

Comment:
Worth mentioning the nf-core extension pack for other goodies?

Reply:
Should be an h3 nested under VSCode? Or be called "VS Code Extensions"

Reply:
Nice. I'll add a section for nf-core and I'll update the title to "VS Code Extensions" to improve clarity and prevent too much nesting.